### PR TITLE
Redo Android KBs

### PR DESCRIPTION
**Pull request by BlueSquare1**

### CODE REVIEW

File: MOBILE_CLIENT/ANDROID/_MEDIUM/UNTRUSTED_EXTERNAL_STORAGE_ACCESS/description.md
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_MEDIUM/UNTRUSTED_EXTERNAL_STORAGE_ACCESS/description.md:1]: This one should stay.

File: MOBILE_CLIENT/ANDROID/_HARDENING/APK_INSECURE_NETWORK_SECURITY_CONFIG/recommendation.md [hunk 1]
```diff
@@ -1,3 +1,47 @@
-It is strongly recommended to add a Network Security Configuration settings enabling encrypted traffic only.
+Android Network Security Configuration enables a declarative setting of the application network security.
 
-Use of a custom Certificate Authority with certificate pinning increases the security of you application and is recommended for security sensitive application.
\ No newline at end of file
+The features enable configuring:
+
+* Custom Certificate Authority with support for debug only settings
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_HARDENING/APK_INSECURE_NETWORK_SECURITY_CONFIG/recommendation.md:5]: Not actionable. What should I do?

File: MOBILE_CLIENT/ANDROID/_HARDENING/APK_NOT_OBFUSCATED/recommendation.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-Design the application to add the following protections and slow reverse engineering of the application:
+Design the application with the following protections to make it harder to reverse engineer it:
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_HARDENING/APK_NOT_OBFUSCATED/recommendation.md:1]: Design?

Not actionable, what should I do?

File: MOBILE_CLIENT/ANDROID/_LOW/APK_LIST_NOT_USED_PERMISSIONS/recommendation.md [hunk 3]
```diff
@@ -1 +1 @@
-Remove unused permissions.
\ No newline at end of file
+Remove unused declared permissions from the application's manifest
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_LOW/APK_LIST_NOT_USED_PERMISSIONS/recommendation.md:1]: Please provide a code example.

File: MOBILE_CLIENT/ANDROID/_LOW/INTENT_SPOOFING/recommendation.md [hunk 4]
```diff
@@ -1,11 +1,12 @@
-To limit one's exposure to this type of attack, developers should avoid exporting components unless the component is
-specifically designed to handle requests from untrusted applications. Developers should be aware that declaring an
-intent filter will automatically export the component, exposing it to public access. Critical, state-changing actions
-should not be placed in exported components. If a single component handles both inter- and intra-application
-requests, the developer should consider dividing that component into separate components.
+To limit one's exposure to this type of attack, developers should:
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_LOW/INTENT_SPOOFING/recommendation.md:1]:
```suggestion
To limit one's exposure to this type of attack, consider the following recommendations: 
```

File: MOBILE_CLIENT/ANDROID/_LOW/INTENT_SPOOFING/recommendation.md [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
-requests, the developer should consider dividing that component into separate components.
+To limit one's exposure to this type of attack, developers should: 
 
-If a component must be exported (e.g., to receive system broadcasts), then the component should dynamically check the
-caller's identity prior to performing any operations. Requiring Signature or SignatureOrSystem permissions is an
-effective way of limiting a component's exposure to a set of trusted applications. Finally, the return values of
-exported components can also leak private data, so developers should check the caller's identity prior to returning
-sensitive values.
+- Avoid exporting components unless the component is
+specifically designed to handle requests from untrusted applications.
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_LOW/INTENT_SPOOFING/recommendation.md:4]: Don't break line on full sentences. It breaks in the UI.

File: MOBILE_CLIENT/ANDROID/_LOW/PRIVACY_KEYBOARD_CACHE/recommendation.md [hunk 6]
```diff
@@ -0,0 +1 @@
+Sensitive input fields should be marked as `password` input field to avoid having sensitive data logged into keyboard cache.
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_LOW/PRIVACY_KEYBOARD_CACHE/recommendation.md:1]: Need a better recommendation.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: MOBILE_CLIENT/ANDROID/_HARDENING/APK_ELF_NOTPROTECTED/recommendation.md [hunk 7]
```diff
@@ -1,5 +1,5 @@
-If `Stack canary` is missing, Compilers such as GCC enable this feature if requested through compiler options:
+To ensure that the Stack canary feature is enabled when compiling with GCC, you can specify one of the compiler options:
```
Comment (benyissa) [inline MOBILE_CLIENT/ANDROID/_HARDENING/APK_ELF_NOTPROTECTED/recommendation.md:1]:
```suggestion
To ensure that the stack canary feature is enabled when compiling with GCC, you can specify one of the compiler options:
```

File: MOBILE_CLIENT/ANDROID/_HARDENING/APK_INSECURE_NETWORK_SECURITY_CONFIG/recommendation.md [hunk 8]
```diff
@@ -1,3 +1,47 @@
-It is strongly recommended to add a Network Security Configuration settings enabling encrypted traffic only.
+Android Network Security Configuration enables a declarative setting of the application network security.
```
Comment (benyissa) [inline MOBILE_CLIENT/ANDROID/_HARDENING/APK_INSECURE_NETWORK_SECURITY_CONFIG/recommendation.md:1]:
```suggestion
Android Network Security Configuration enables a declarative setting of the application's network security.
```

**Review by amine3 — DISMISSED** [review]

**Review by ErebusZ — COMMENTED** [review]
it might be worth splitting this PR.

File: MOBILE_CLIENT/ANDROID/_MEDIUM/UNTRUSTED_EXTERNAL_STORAGE_ACCESS/description.md
Comment (BlueSquare1) [reply MOBILE_CLIENT/ANDROID/_MEDIUM/UNTRUSTED_EXTERNAL_STORAGE_ACCESS/description.md:1]: I replaced it with `INSECURE_STORAGE` KB

File: MOBILE_CLIENT/ANDROID/_HARDENING/APK_INSECURE_NETWORK_SECURITY_CONFIG/description.md [hunk 9]
```diff
@@ -1,47 +1,4 @@
 Android Network Security Configuration enables a declarative setting of the application network security.
 
-The features enable configuring:
+When Android Network Security Configuration is missing or configured in an insecure way, it can leave application vulnerable to MiTM.
 
-* Custom Certificate Authority with support for debug only settings
```
[the remark this answers is not in the document]
Comment (BlueSquare1) [reply MOBILE_CLIENT/ANDROID/_HARDENING/APK_INSECURE_NETWORK_SECURITY_CONFIG/description.md:5]: This is a recommendation code that was put in the vulnerability description

**Review by oussamaessaji — APPROVED** [review]

**Comment by 3asm** [comment]
@oussamaessaji please stop approving PRs without proper review.

**Comment by 3asm** [comment]
@BlueSquare1 what is the status of this PR?

**Comment by BlueSquare1** [comment]
> @BlueSquare1 what is the status of this PR?

All comments have been addressed, other KB categories will have separate PRs.

**Review by 3asm — APPROVED** [review]

**Pull request merged**